### PR TITLE
use spans instead of tracers

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
Using spans everywhere, I think logically makes more sense and avoiding using Thread.current gives us a good performance boost.

We went from 100ns in the JSON rack middleware down to 28ns.

@jfeltesse-mdsol  @adriancole @ykitamura-mdsol 
tell me what do you think of this approach

### CODE REVIEW

File: lib/zipkin-tracer/tracer_factory.rb [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+
+    def flush!
+      @scribe.batch do
+        messages = spans.values.map do |span|
+          buf = ''
+          trans = Thrift::MemoryBufferTransport.new(buf)
+          oprot = Thrift::BinaryProtocol.new(trans)
+          span.to_thrift.write(oprot)
+          binary = Base64.encode64(buf).gsub("\n", "")
+          @scribe.log(binary, TRACER_CATEGORY)
```
Comment (adriancole) [inline lib/zipkin-tracer/tracer_factory.rb]: I think you want to make one call to log here.. it appears like this block makes a call for each span. scribe can accept a list of log entries in a single network call.

File: lib/zipkin-tracer/zipkin_scribe_tracer.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    def flush!
+      @scribe.batch do
+        messages = spans.values.map do |span|
+          buf = ''
+          trans = Thrift::MemoryBufferTransport.new(buf)
+          oprot = Thrift::BinaryProtocol.new(trans)
+          span.to_thrift.write(oprot)
+          binary = Base64.encode64(buf).gsub("\n", "")
+          @scribe.log(binary, TRACER_CATEGORY)
```
Comment (adriancole) [inline lib/zipkin-tracer/zipkin_scribe_tracer.rb]: same comment

**Comment by adriancole** [comment]
overall comment that adding to a span is more explicit, and more like work going on in opentracing. Sound like a positive move.

**Comment by ykitamura-mdsol** [comment]
LGTM!

Can I use the "with_new_span" method from local tracer like this?

```ruby
Trace.tracer.with_new_span(Trace.id, name) do |span|
```

**Pull request merged**